airflow/models.py

```diff
@@ -2058,6 +2068,9 @@ class derived from this one results in the creation of a task object,
     :type resources: dict
     :param run_as_user: unix username to impersonate while running the task
     :type run_as_user: str
+    :param task_concurrency: When set, a task will be able to limit the concurrent
```

Comment:
We should probably rename concurrency to dag_concurrency to make the distinction clear. At the least probably the variable names should be changed.

Reply:
Seems kinda shitty since it's already named. I'm not sure if renaming it internal-only is that helpful.

Reply:
Having both concurrency and task_concurrency is pretty confusing (especially when users already get confused between thinks like concurrency and dagruns), I think it's worth the cost of changing the name. We can deprecate the old one. I get that it's a big chunk of work though so I'll leave it up to you, not a blocker.

Reply:
The problem is it is also a breaking change

Reply:
And I don't think we can get this into 1.9

Reply:
We should deprecate it (i.e. take 'concurrency' in user dags and change it to 'task_concurrency' with a deprecation warning).
It's OK if we don't get this into 1.9.